metadata-ingestion/tests/integration/db2/test_db2.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+import re
+import subprocess
+
+import pytest
+import sqlalchemy
+import yaml
+
+from datahub.ingestion.run.pipeline import Pipeline
+from datahub.testing import mce_helpers
+from tests.test_helpers.docker_helpers import wait_for_port
+
+logger = logging.getLogger(__name__)
+
+DB2_PORT = 50000
+
+
+@pytest.fixture(scope="module")
+def test_resources_dir(pytestconfig):
+    return pytestconfig.rootpath / "tests/integration/db2"
+
+
+def is_db2_up(container_name: str) -> bool:
+    cmd = f"docker logs {container_name} 2>&1 | grep 'Setup has completed.'"
+    ret = subprocess.run(
+        cmd,
+        shell=True,
+    )
+    return ret.returncode == 0
```

Comment:
I usually prefer to push down this to the docker infra level.
You can define a healthcheck in the docker compose to determine the service is healthy and so you don't need to handle this in the app layer